### PR TITLE
Program with arguments for CREATE

### PR DESCRIPTION
**Pull request by lukasz-glen**

### CODE REVIEW

File: src/program_generator/pg_arguments.py [hunk 1]
```diff
@@ -108,6 +108,23 @@ def _do_generate(self, opcode, count, op_count):
 
   def _generate_program_triplet(self, operation, op_count):
     opcode = operation['Mnemonic']
+
+    # the program triplet will be for the following number of measured OPCODEs
+    op_counts = [0, op_count, op_count * 2]
+
+    if opcode == "CREATE":
```
Comment (pgebal) [inline src/program_generator/pg_arguments.py:115]: To have it generated by default we need to add CREATE opcode to `selection_arguments.csv`

File: src/program_generator/pg_arguments.py [hunk 2]
```diff
@@ -108,6 +108,23 @@ def _do_generate(self, opcode, count, op_count):
...
+      opcode_arity = 3
+      mem_allocation = "5f61080052"
+      initial_code_on_stack = "6c6460016001015f526005601bf360981b"
+      initial_code_in_mem = ""
+      for pre_offset in range(32):
+        offset = "%0.4X" % (32 * pre_offset)
```
Comment (pgebal) [inline src/program_generator/pg_arguments.py:121]: Magic. I didn't even know that you can create hex numbers that way.

File: src/program_generator/pg_arguments.py [hunk 3]
```diff
@@ -108,6 +108,23 @@ def _do_generate(self, opcode, count, op_count):
 
   def _generate_program_triplet(self, operation, op_count):
     opcode = operation['Mnemonic']
+
+    # the program triplet will be for the following number of measured OPCODEs
+    op_counts = [0, op_count, op_count * 2]
+
+    if opcode == "CREATE":
+      opcode_arity = 3
```
Comment (pgebal) [inline src/program_generator/pg_arguments.py:116]: Unused.

File: src/program_generator/pg_arguments.py [hunk 4]
```diff
@@ -108,6 +108,23 @@ def _do_generate(self, opcode, count, op_count):
...
+      mem_allocation = "5f61080052"
+      initial_code_on_stack = "6c6460016001015f526005601bf360981b"
+      initial_code_in_mem = ""
+      for pre_offset in range(32):
+        offset = "%0.4X" % (32 * pre_offset)
+        initial_code_in_mem += "8061" + offset + "52"
+      prefix_code = mem_allocation + initial_code_on_stack + initial_code_in_mem
+      arg_sizes = [1, random.randint(1, 32), random.randint(1, 32)]
+      single_op_pushes = ["61%0.4X" % (arg_sizes[2] * 32), "61%0.4X" % ((arg_sizes[1] - 1) * 32), "5f"]
```
Comment (pgebal) [inline src/program_generator/pg_arguments.py:125]: This is not obvious.
Please leave a descriptive comment explaining structure of the full code for CREATE,
explain why there is `SHL` used, what's in memory, etc.

Comment (lukasz-glen) [reply src/program_generator/pg_arguments.py:133]: code is changed a little to be more straight forward

File: src/program_generator/pg_arguments.py (hunk 1 above)
Comment (lukasz-glen) [reply src/program_generator/pg_arguments.py:115]: This is for another PR. Maybe we should split it into several files and coordinate with marginal program generation. Not sure yet.

**Review by JacekGlen — APPROVED** [review]

**Pull request merged**